l1-contracts/contracts/zksync/facets/Executor.sol

```diff
@@ -472,4 +496,90 @@ contract ExecutorFacet is Base, IExecutor {
     function _setBit(uint256 _bitMap, uint8 _index) internal pure returns (uint256) {
         return _bitMap | (1 << _index);
     }
+
+    /// @notice Calls the point evaluation precompile and verifies the output
+    /// Verify p(z) = y given commitment that corresponds to the polynomial p(x) and a KZG proof.
+    /// Also verify that the provided commitment matches the provided versioned_hash.
+    /// 
+    function _pointEvaluationPrecompile(
+        bytes32 _versionedHash,
+        bytes32 _openingPoint,
+        bytes calldata _openingValueCommitmentProof
+    ) internal view {
+        bytes memory precompileInput = abi.encodePacked(_versionedHash, _openingPoint, _openingValueCommitmentProof);
+
+        (bool success, bytes memory data) = POINT_EVALUATION_PRECOMPILE_ADDR.staticcall(precompileInput);
+
+        require(success, "failed to call point evaluation precompile");
+        (, uint256 result) = abi.decode(data, (uint256, uint256));
+        require(result == BLS_MODULUS, "precompile unexpected output");
+    }
+
+    /// Verifies that the blobs contain the correct data by calling the point evaluation precompile. For the precompile we need:
+    /// versioned hash || opening point || opening value || commitment || proof
+    /// the _pubdataCommitments will contain the last 4 values, the versioned hash is pulled from the BLOBHASH opcode
+    /// pubdataCommitments is a list of: opening point (16 bytes) || claimed value (32 bytes) || commitment (48 bytes) || proof (48 bytes)) = 144 bytes
+    function _verifyBlobInformation(
+        bytes calldata _pubdataCommitments,
+        bytes32[2] memory _blobHashes
+    ) internal view returns (bytes32[] memory blobCommitments) {
+        uint256 versionedHashIndex = 0;
+
+        // TODO: This should be dynamic instead of being hardcoded to 2 blobs
+        require(_pubdataCommitments.length > 0, "pl");
+        require(
+            _pubdataCommitments.length <= PUBDATA_COMMITMENT_SIZE * 2 &&
+                _pubdataCommitments.length % PUBDATA_COMMITMENT_SIZE == 0,
+            "bs"
+        );
+        blobCommitments = new bytes32[](2);
+        bytes32 versionedHash;
+
+        for (uint256 i = 0; i < _pubdataCommitments.length; i += PUBDATA_COMMITMENT_SIZE) {
+            versionedHash = _getBlobVersionedHash(versionedHashIndex);
+
+            require(versionedHash != bytes32(0), "vh");
+
+            // First 16 bytes is the opening value. While we get the value as 16 bytes, the point evaluation precompile
+            // requires it to be 32 bytes. The blob commitment must use the opening value as 16 bytes though.
+            bytes32 openingValue = bytes32(uint256(uint128(bytes16(_pubdataCommitments[i: i + PUBDATA_COMMITMENT_CLAIMED_VALUE_OFFSET]))));
```

Comment:
Does that mean value or opening point? If value - it's always 32 bytes, if point - it's 128 bits

Reply:
It should be the opening point